spring-cloud-task-core/src/main/java/org/springframework/cloud/task/configuration/TaskLifecycleConfiguration.java

```diff
@@ -62,17 +62,27 @@
 
 	private boolean initialized = false;
 
+	private io.micrometer.core.instrument.MeterRegistry meterRegistry;
+
+	private ObservationRegistry observationRegistry;
+
 	@Autowired
 	public TaskLifecycleConfiguration(TaskProperties taskProperties,
-			ConfigurableApplicationContext context, TaskRepository taskRepository,
-			TaskExplorer taskExplorer, TaskNameResolver taskNameResolver,
-			ObjectProvider<ApplicationArguments> applicationArguments) {
+		ConfigurableApplicationContext context, TaskRepository taskRepository,
+		TaskExplorer taskExplorer, TaskNameResolver taskNameResolver,
+		ObjectProvider<ApplicationArguments> applicationArguments,
+		@Autowired(required = false) io.micrometer.core.instrument.MeterRegistry meterRegistry,
+		@Autowired(required = false) ObservationRegistry observationRegistry) {
```

Comment:
You can do a `ObservationRegistry.NOOP` instead of setting `null`. That way you won't need any `null` checks.